pennylane/_qubit_device.py

```diff
@@ -136,6 +140,34 @@ def __init__(self, wires=1, shots=None, cache=0, analytic=None):
         """OrderedDict[int: Any]: Mapping from hashes of the circuit to results of executing the
         device."""
 
+        self.custom_decomps = custom_decomps
+
+        if custom_decomps:
+            # Get a list of all the operations requiring custom decomps
+            custom_op_names = [
+                op if isinstance(op, str) else op.__name__ for op in custom_decomps.keys()
+            ]
+
+            custom_decomp_condition =  qml.BooleanFn(
+                lambda obj: not isinstance(obj, qml.tape.QuantumTape) # Expand templates
+                and obj.name not in custom_op_names # Expand things that don't have custom decomp
+                and self.supports_operation(obj.name) # Expand things until supported on device
+            )
```

Comment:
@glassnotes did 
```python
custom_decomp_condition = qml.BooleanFn(lambda obj: obj.name not in custom_op_names)
custom_decomp_condition &= self.stopping_condition
```
not work?

Reply:
this is the same as the following:
```python
condition = qml.BooleanFn(lambda obj: obj.name not in custom_op_names)
custom_fn = qml.transforms.create_expand_fn(depth=10, stop_at=condition, device=self)

Reply:
Actually I had a weird amount of trouble with the stopping condition.  It seemed like if I did not include that first condition there, it would not expand the templates :confused: Let me give this another go, though.

Reply:
oh, the first one is included within the devices stopping condition though 🤔 

![image](https://user-images.githubusercontent.com/2959003/142210970-df1d03dc-da0b-4967-bf73-633ed1219e93.png)

Reply:
Ahhh I found the problem I had originally. I was including both `obj.name not in custom_op_names` and 
`self.supports_operation(obj.name)` in the stopping condition I was passing to the `create_expand_fn`, and the second one was being cancelled out in the `&=` with the device criteria. Thanks!